fv3core/stencils/mapn_tracer.py

```diff
@@ -7,31 +7,68 @@
 from fv3core.utils.typing import FloatField
 
 
-def compute(
-    pe1: FloatField,
-    pe2: FloatField,
-    dp2: FloatField,
-    tracers: Dict[str, "FloatField"],
-    nq: int,
-    q_min: float,
-    i1: int,
-    i2: int,
-    j1: int,
-    j2: int,
-    kord: int,
-):
-    qs = utils.make_storage_from_shape(
-        pe1.shape, origin=(0, 0, 0), cache_key="mapn_tracer_qs"
-    )
-    map_single = utils.cached_stencil_class(MapSingle)(
-        kord, 0, i1, i2, j1, j2, cache_key=f"mapntracer-single-j{j2}"
-    )
-
-    for q in utils.tracer_variables[0:nq]:
-        map_single(tracers[q], pe1, pe2, qs)
-
-    if spec.namelist.fill:
-        fillz = utils.cached_stencil_class(FillNegativeTracerValues)(
-            cache_key="mapntracer-fillz"
+class MapNTracer:
+    """
+    Fortran code is mapn_tracer, test class is MapN_Tracer_2d
+    """
+
+    def __init__(self, kord: int, i1: int, i2: int, j1: int, j2: int):
+        grid = spec.grid
+        self._origin = (i1, j1, 0)
+        self._domain = ()
+        self._nk = grid.npz
+        self._i1 = i1
+        self._i2 = i2
+        self._j1 = j1
+        self._j2 = j2
+        self._qs = utils.make_storage_from_shape(
+            (grid.npx, grid.npy, self._nk + 1),
+            origin=(0, 0, 0),
+            cache_key="mapn_tracer_qs",
+        )
+
+        self._map_single = utils.cached_stencil_class(MapSingle)(
```

Comment:
Do we still need cache keys for storages and stencils that are object members?

Reply:
Nope, I just neglected to update that